File: tests/bench/mandelbrot.py
import os

_LEVEL = os.environ.get("BENCH_LEVEL", "hard")
# Width and height are image dimensions: work is their product, so the level
# halves each rather than dividing by 16 -- halving both is already 4x less.
DIM_DIV = 4 if _LEVEL == "easy" else 2 if _LEVEL == "medium" else 1

width = 3200 // DIM_DIV
height = 240 // DIM_DIV
limit = 100

inside = 0
for py in range(height):
    y0 = float(py) * 2.0 / float(height) - 1.0
    for px in range(width):
        x0 = float(px) * 3.0 / float(width) - 2.0
        x = 0.0
        y = 0.0
        i = 0
        while i < limit:
            x2 = x * x
            y2 = y * y
            if x2 + y2 > 4.0:
                break
            xy = x * y
            y = 2.0 * xy + y0
            x = x2 - y2 + x0
            i += 1
        if i == limit:
            inside += 1
print(limit)
print(inside)

File: tests/bench/Mandelbrot.java
// Same program as mandelbrot.jai.
public class Mandelbrot {
    public static void main(String[] args) {
        final int width = 3200;
        final int height = 240;
        final int limit = 100;

        long inside = 0;
        for (int py = 0; py < height; py++) {
            double y0 = (double) py * 2.0 / (double) height - 1.0;
            for (int px = 0; px < width; px++) {
                double x0 = (double) px * 3.0 / (double) width - 2.0;
                double x = 0.0;
                double y = 0.0;
                int i = 0;
                while (i < limit) {
                    double x2 = x * x;
                    double y2 = y * y;
                    if (x2 + y2 > 4.0) break;
                    double xy = x * y;
                    y = 2.0 * xy + y0;
                    x = x2 - y2 + x0;
                    i++;
                }
                if (i == limit) inside++;
            }
        }
        System.out.println(limit);
        System.out.println(inside);
    }
}
